libs/database-entities/src/main/java/gov/cdc/nbs/patient/PatientNameHistoryListener.java

```diff
@@ -0,0 +1,35 @@
+package gov.cdc.nbs.patient;
+
+import gov.cdc.nbs.entity.odse.Person;
+import gov.cdc.nbs.entity.odse.PersonName;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.PreUpdate;
+import javax.transaction.Transactional;
+
+@Component
+public class PatientNameHistoryListener {
+    private PatientNameHistoryCreator creator;
+    private final JdbcTemplate template;
+
+    public PatientNameHistoryListener(final PatientNameHistoryCreator creator, JdbcTemplate template) {
+        this.creator = creator;
+        this.template = template;
+    }
+
+    @PreUpdate
+    void preUpdate(PersonName personName) {
+        int personNameSequence = personName.getId().getPersonNameSeq();
+        int currentVersion = getCurrentVersionNumber(personName.getPersonUid(), personNameSequence);
+        this.creator.createPersonNameHistory(personName.getPersonUid().getId(), currentVersion + 1, personNameSequence);
+    }
+
+    @Transactional
```

Comment:
The `@Transactional` annotation isn't doing anything here.  It needs to be on a public method so that a proxy class can be created to ensure that a transaction is open.  It also doesn't work the same when calling a `@Transactional` method within the same class, it would need to be called from an injected object so Spring can wrap it in a proxy.

Reply:
Agreed! I'll remove it in the next PR.